src/Adaptive.Agrona.Tests/AtomicBufferTests.cs

```diff
@@ -38,14 +41,58 @@ public class AtomicBufferTests
         private const long LongValue = int.MaxValue + 5L;
         private const double DoubleValue = int.MaxValue + 7.0d;
 
-        [Datapoint] public readonly IAtomicBuffer ByteArrayBacked = new UnsafeBuffer(new byte[BufferCapacity], 0, BufferCapacity);
-        [Datapoint] public readonly IAtomicBuffer AlignedByteArrayBacked = new UnsafeBuffer(BufferUtil.AllocateDirectAligned(BufferCapacity, BitUtil.CACHE_LINE_LENGTH));
+        private string TempFileName;
+        private MappedByteBuffer MappedByteBuffer;
 
-        [Datapoint] public static readonly IAtomicBuffer UnmanagedBacked = new UnsafeBuffer(Marshal.AllocHGlobal(BufferCapacity), BufferCapacity);
+        private IAtomicBuffer ByteArrayBacked;
+        private IAtomicBuffer AlignedByteArrayBacked;
+
+        private IAtomicBuffer UnmanagedBacked;
 
-        private static readonly MappedByteBuffer MappedByteBuffer = new MappedByteBuffer(MemoryMappedFile.CreateNew("testmap", BufferCapacity));
+        private IAtomicBuffer MemoryMappedFileBacked;
+        private bool Initialized;
 
-        [Datapoint] public static readonly IAtomicBuffer MemoryMappedFileBacked = new UnsafeBuffer(MappedByteBuffer.Pointer, BufferCapacity);
+        public void Initialize()
```

Comment:
clearer to have a [OneTimeSetup] and remove the Initialized flag and call from Datapoints()?

Reply:
I thought the same.  Unfortunately (and I tried all kinds of ways to do this), `[OneTimeSetup]` does not run before the `Datapoints` are evaluated.  I tried all kinds of different orders/things and this was the only way that worked.  Happy to re-evaluate if there's another way.

Reply:
Fair enough - that's not really intuitive behaviour from nunit.